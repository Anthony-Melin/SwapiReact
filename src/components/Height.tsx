import { PropsWithChildren } from 'react'

type HeightProps = PropsWithChildren<{ value: string }>

const Height = ({ value, ...restProps }: HeightProps) => {
    return (
        <p {...restProps}>
            Height: {value}
        </p>
    )
}

export default Height
