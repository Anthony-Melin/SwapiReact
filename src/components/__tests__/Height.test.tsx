import { describe, it, expect, beforeEach } from 'vitest'
import { render, screen } from '@testing-library/react'
import Height from '../Height.tsx'

describe('Height', () => {
    beforeEach(() => {
        render(<Height value="VALUE" />)
    })

    it('should match snapshot', () => {
        expect(screen.getByTestId('height')).toMatchSnapshot()
    })

    it('should have text', () => {
        expect(screen.getByTestId('height')).toHaveTextContent(
            'Height: VALUE'
        )
    })
})
