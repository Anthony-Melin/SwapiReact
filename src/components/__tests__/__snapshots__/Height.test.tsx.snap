// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Height > should match snapshot 1`] = `
<p
  data-testid="height"
>
  Height
  : 
  VALUE
</p>
`;
